test/unit/history/map.test.js

```diff
@@ -0,0 +1,17 @@
+import History from '../../../src/history'
+
+describe('History::map', function() {
+  const action = n => n
+
+  it('provides the index', function() {
+    const history = new History({ maxHistory: Infinity })
+
+    history.append(action, 'resolve')
+    history.append(action, 'resolve')
+    history.append(action, 'resolve')
+
+    let result = history.map((action, i) => i)
+
+    expect(result).toEqual([0, 1, 2, 3])
```

Comment:
Yeah. This gets back to another discussion we never finished: should we ignore internal actions like `birth` and `start`?